pkg/daemon/rpm-ostree.go

```diff
@@ -164,87 +154,41 @@ func (r *RpmOstreeClient) PullAndRebase(container string, keep bool) (imgid stri
 		authArgs = append(authArgs, "--authfile", kubeletAuthFile)
 	}
 
-	// If we're passed a non-canonical image, resolve it to its sha256 now
-	isCanonicalForm := true
-	if _, err = getRefDigest(container); err != nil {
-		isCanonicalForm = false
-		// In non-canonical form, we pull unconditionally right now
-		args := []string{"pull", "-q"}
-		args = append(args, authArgs...)
-		args = append(args, container)
-		_, err = pivotutils.RunExt(numRetriesNetCommands, "podman", args...)
-		if err != nil {
-			return
-		}
-	} else {
-		if previousPivot != "" {
-			var targetMatched bool
-			targetMatched, err = compareOSImageURL(previousPivot, container)
-			if err != nil {
-				return
-			}
-			if targetMatched {
-				changed = false
-				return
-			}
-		}
-
-		// Pull the image
-		args := []string{"pull", "-q"}
-		args = append(args, authArgs...)
-		args = append(args, container)
-		_, err = pivotutils.RunExt(numRetriesNetCommands, "podman", args...)
-		if err != nil {
-			return
-		}
-	}
+	inspectArgs := []string{"inspect"}
+	inspectArgs = append(inspectArgs, authArgs...)
 
-	inspectArgs := []string{"inspect", "--type=image"}
-	inspectArgs = append(inspectArgs, fmt.Sprintf("%s", container))
+	// Map container image name with transport so that skopeo understands it
+	imageName := "docker://" + container
+	inspectArgs = append(inspectArgs, imageName)
 	var output []byte
-	output, err = runGetOut("podman", inspectArgs...)
+	output, err = runGetOut("skopeo", inspectArgs...)
 	if err != nil {
 		return
 	}
-	var imagedataArray []imageInspection
-	err = json.Unmarshal(output, &imagedataArray)
+	var imagedata imageInspection
+	err = json.Unmarshal(output, &imagedata)
 	if err != nil {
-		err = errors.Wrapf(err, "unmarshaling podman inspect")
+		err = errors.Wrapf(err, "unmarshaling skopeo inspect")
 		return
 	}
-	imagedata := imagedataArray[0]
+
+	// If we're passed a non-canonical image, resolve it to its sha256 now
+	isCanonicalForm := true
+	if _, err = getRefDigest(container); err != nil {
+		isCanonicalForm = false
+	}
+
+	var imgid string
 	if !isCanonicalForm {
 		imgid = imagedata.RepoDigests[0]
 		glog.Infof("Resolved to: %s", imgid)
 	} else {
 		imgid = container
 	}
 
-	containerName := pivottypes.PivotNamePrefix + string(uuid.NewUUID())
-
-	// `podman mount` wants a container, so let's make create a dummy one, but not run it
-	var cidBuf []byte
-	cidBuf, err = runGetOut("podman", "create", "--net=none", "--annotation=org.openshift.machineconfigoperator.pivot=true", "--name", containerName, imgid)
-	if err != nil {
-		return
-	}
-	defer func() {
-		// Kill our dummy container
-		podmanRemove(containerName)
-	}()
-
-	cid := strings.TrimSpace(string(cidBuf))
-	// Use the container ID to find its mount point
-	var mntBuf []byte
-	mntBuf, err = runGetOut("podman", "mount", cid)
```

Comment:
Aren't we missing a call to the extract function here?

Reply:
Hopefully I understood your question correctly.
We are now not mounting container image. https://github.com/openshift/machine-config-operator/pull/1941/files#diff-06961b075f1753956d802ba954d2cfb5R251 is where we extract OS container content in /run/mco-machine-os-content/ and use same for rpm-ostree rebase.